libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/ElapsedTime.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.services.android.navigation.v5.exception.NavigationException
+import kotlin.math.roundToLong
+
+internal class ElapsedTime {
+    var start: Long? = null
+        private set
+    var end: Long? = null
+        private set
+
+    companion object {
+        private const val ELAPSED_TIME_DENOMINATOR = 1e+9
+        private const val PRECISION = 100.0
+    }
+
+    val elapsedTime: Double
+        get() {
+            if (start == null || end == null) {
+                throw NavigationException("Must call start() and end() before calling getElapsedTime()")
+            }
+            val elapsedTimeInNanoseconds = end!! - start!!
```

Comment:
please avoid use `!!`

Reply:
Yeah, I know it bad practice... but it allowing to use in very rare cases, for example like this. To avoid using !! in this piece of code, we need to add additional nested checks like 
```suggestion
            start?.let{ startTime -> 
                end?.let{ endTime -> 
                    ...
                }
            }
```
But I think it makes readability of this method much worth... 

Or we can add additional 
```suggestion
            val startTime = start ?: throw Exception("...")
            val endTime = end ?: throw Exception("...")
```
But I think it also not the best solution. What do you think?

Reply:
I'm gonna to write a few `nonNull` extensions. please wait for them

Reply:
PTAL https://github.com/mapbox/mapbox-navigation-android/pull/2098